File: repaso objetos/java/objetos/java/instrumentos/Banda.java
package objetos.java.instrumentos;

import java.util.ArrayList;
import java.util.Random;

public class Banda {
    ArrayList<Musico> musicos = new ArrayList<Musico>();

    public void agregarMusico(String nombre){
        Musico m=new Musico();
        m.setNombre(nombre);
        musicos.add(m);
    }

    private Instrumento generarInstrumento(){

        Random rn = new Random();
        int opc = rn.nextInt(3);
        switch(opc){
            case 0:
                return new Guitarra();
            case 1:
                return new Bajo();
            default:
                return new Violin();
        }
    }
    public void presentarBanda(){
        for (Musico musico : musicos){
            musico.presentar();
            musico.tocar(generarInstrumento());
            
        }
    }
}

File: repaso objetos/python/PhytonEst copy/Banda.py
from Musico import Musico
from Clases.Instrumento import *
import random

class Banda():
    
    def __init__(self):
        self.musicos=list()

    def agregarMusico(self,nombre):
        m=Musico()
        m.set__nombre__(nombre)
        self.musicos.append(m)  

    def generarInstrumento(self):
        rn = random.randint(0,3)
        sel = rn
        i=Instrumento()
        if sel == 1:
            i= Guitarra()
            return i
        elif sel == 2:
            i= Bajo()
            return i
        else:
            i= Violin()
            return i
        return i   

    def presentarBanda(self):
        for i in self.musicos: 
            i.presentar()
            i.tocar(Violin())
            
            
               






